### PR TITLE
Adding check and show error message in on-prem scenario

### PR DESCRIPTION
**Pull request by jahsu-MSFT**

### CODE REVIEW

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs [hunk 1]
```diff
@@ -69,6 +69,12 @@ public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
 
             string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
 
+            bool onPrem = !String.Equals(context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.ServerType)?.Value, "Hosted", StringComparison.OrdinalIgnoreCase);
+            if(onPrem)
```
Comment (fadnavistanmay) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs]: nit: if (onPrem) - note: space after if.

File: src/Misc/layoutbin/en-US/strings.json [hunk 2]
```diff
@@ -389,6 +389,7 @@
   "NotLinux": "This Agent version is built for Linux. Please install a correct build for your OS.",
   "NotOSX": "This Agent version is built for OSX. Please install a correct build for your OS.",
   "NotWindows": "This Agent version is built for Windows. Please install a correct build for your OS.",
+  "OnPremIsNotSupported": "Pipieline Artifact Task is not supported in on-premises. Please use Build Artifact Task instead.",
```
Comment (fadnavistanmay) [inline src/Misc/layoutbin/en-US/strings.json]: Pipeline*

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 3]
```diff
@@ -76,6 +77,12 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
 
             targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
 
+            bool onPrem = !String.Equals(context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.ServerType)?.Value, "Hosted", StringComparison.OrdinalIgnoreCase);
+            if(onPrem)
```
Comment (fadnavistanmay) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: if (onPrem) - same as above.

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs (hunk 1 above)
Comment (jahsu-MSFT) [reply src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs]: updated

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 4]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.VisualStudio.Services.WebApi;
 using Microsoft.VisualStudio.Services.Agent.Util;
 using Agent.Sdk;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
```
Comment (srdorbal) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: sort them

**Comment by jahsu-MSFT** [comment]
@TingluoHuang Please merge the PR whenever you're ready.

**Comment by jtpetty** [comment]
This PR passed on requeue. https://dev.azure.com/mseng/PipelineTools/_build/results?buildId=10384206&view=results

**Pull request merged**